QgisModelBaker/gui/generate_project.py

```diff
@@ -190,10 +191,37 @@ def accepted(self):
         configuration.dbschema = configuration.dbschema or configuration.database
         self.save_configuration(configuration)
 
-        # create schema with superuser
         db_factory = self.db_simple_factory.create_factory(db_id)
-        res, message = db_factory.pre_generate_project(configuration)
 
+        try:
+            # raise warning when the schema or the database file already exists
+            config_manager = db_factory.get_db_command_config_manager(configuration)
+            db_connector = db_factory.get_db_connector(
+                config_manager.get_uri(configuration.db_use_super_login) or config_manager.get_uri(), configuration.dbschema)
+
+            if db_connector.db_or_schema_exists():
+                if interlis_mode:
+                    warning_box = QMessageBox(self)
+                    warning_box.setIcon(QMessageBox.Warning)
+                    warning_title = self.tr("{} already exists").format(
+                        db_factory.get_specific_messages()['db_or_schema']
+                    ).capitalize()
+                    warning_box.setWindowTitle(warning_title)
+                    warning_box.setText(self.tr("{db_or_schema}: {db_or_schema_name}\n\nDo you want to "
+                                                "import anyway?").format(
+                        db_or_schema=db_factory.get_specific_messages()['db_or_schema'].capitalize(),
+                        db_or_schema_name=configuration.dbschema or config_manager.get_uri()
+                    ))
+                    warning_box.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+                    warning_box_result = warning_box.exec_()
+                    if warning_box_result == QMessageBox.No:
+                        return
+        except FileNotFoundError:
```

Comment:
Can we deal with this in the gpkg connector?

Reply:
In the GPKGConnectior it raises this `FileNotFoundError`. So everywhere on calling `get_db_connector` it catches this error.
But we could check for the existence of the file in the `gpkg_factory.get_db_connector` and if it does not exist return `None` - so in the GUI classes we check if db_connector is None. What do you think?

Reply:
But anyway. If it would be here it should be `except (DBConnectorError, FileNotFoundError):` since the `FileNotFoundError` is the equivalent of `DBConnectorError`. But anyway. Would the `x_factory.get_db_connector` be the preferred place?

Reply:
The reason why I asked was, if there is a `FileNotFoundError` we expect, should we also expect a `NetworkConnectionFailedError`? And a `SchemaDoesNotExistError`?

But if it's already done like this everywhere, no objections to also do it here.

Reply:
Okay. But thanks for mentioning. Because of your comment I noticed that I forget to catch `DBConnectorError` :-)

Reply:
What my main question is, which errors is this method expected to raise?
If we add more connectors in the future (oracle, ...) do we need to add more catches?

Reply:
These system dependent errors are catched in the connector and then the `DBConnectorError` is raised - so from the outside only the `DBConnectorError` needs to be catched (or the `FileNotFoundError`).